src/ops/git_utils.py

```diff
@@ -0,0 +1,45 @@
+# Copyright 2019 Adobe. All rights reserved.
+# This file is licensed to you under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License. You may obtain a copy
+# of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+# OF ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+import git
+import logging
+import yaml
+
+logger = logging.getLogger(__name__)
+
+
+def setup_repo(repo_path, upstream_repo):
+    """
+    Ensure that the repo is present or clone it from upstream otherwise.
+    """
+    try:
+        git.Repo(repo_path, search_parent_directories=True)
+    except git.NoSuchPathError:
+        logger.warning(
+            "Repo '%s' not found. Cloning from upstream '%s'", repo_path, upstream_repo
+        )
+        git.Repo.clone_from(upstream_repo, repo_path)
+
+
+def checkout_repo(repo_path, config_path, get_version):
+    with open(config_path) as f:
+        conf = yaml.load(f, Loader=yaml.FullLoader)
+
+    version = get_version(conf)
```

Comment:
Where is this method defined?

Reply:
It's given as an argument to the function. It's a way to parameterise the config path for each runner (since they are different between terraform & helmfile) without duplicating the `open` file & `read` data sequence.